src/backend/base/langflow/graph/graph/base.py

```diff
@@ -260,7 +260,7 @@ def add_component(self, component: Component, component_id: str | None = None) -
         if component_id in self.vertex_map:
             msg = f"Component ID {component_id} already exists"
             raise ValueError(msg)
-        frontend_node = component.to_frontend_node()
+        frontend_node = run_until_complete(component.to_frontend_node())
```

Comment:
I think run_until_complete should be avoided as much as possible and only be used in entry points.